tests/MongoDB.Driver.Core.Tests/Core/Authentication/ScramSha1AuthenticatorTests.cs

```diff
@@ -176,5 +174,44 @@ public void Constructor_should_throw_an_ArgumentNullException_when_credential_is
             sentMessages[0].Should().Be("{opcode: \"query\", requestId: " + actualRequestId0 + ", database: \"source\", collection: \"$cmd\", batchSize: -1, slaveOk: true, query: {saslStart: 1, mechanism: \"SCRAM-SHA-1\", payload: new BinData(0, \"biwsbj11c2VyLHI9ZnlrbytkMmxiYkZnT05Sdjlxa3hkYXdM\")}}");
             sentMessages[1].Should().Be("{opcode: \"query\", requestId: " + actualRequestId1 + ", database: \"source\", collection: \"$cmd\", batchSize: -1, slaveOk: true, query: {saslContinue: 1, conversationId: 1, payload: new BinData(0, \"Yz1iaXdzLHI9ZnlrbytkMmxiYkZnT05Sdjlxa3hkYXdMSG8rVmdrN3F2VU9LVXd1V0xJV2c0bC85U3JhR01IRUUscD1NQzJUOEJ2Ym1XUmNrRHc4b1dsNUlWZ2h3Q1k9\")}}");
         }
+
+        [Theory]
+        [ParameterAttributeData]
+        public void Authenticate_should_use_cache(
+            [Values(false, true)] bool async)
+        {
+            var randomStringGenerator = new ConstantRandomStringGenerator("fyko+d2lbbFgONRv9qkxdawL");
+            var subject = new ScramSha1Authenticator(__credential, randomStringGenerator);
+
+            var saslStartReply = MessageHelper.BuildReply<RawBsonDocument>(
+                RawBsonDocumentHelper.FromJson(
+                    "{conversationId: 1, payload: BinData(0,\"cj1meWtvK2QybGJiRmdPTlJ2OXFreGRhd0xIbytWZ2s3cXZVT0tVd3VXTElXZzRsLzlTcmFHTUhFRSxzPXJROVpZM01udEJldVAzRTFURFZDNHc9PSxpPTEwMDAw\"), done: false, ok: 1}"));
+            var saslContinueReply = MessageHelper.BuildReply<RawBsonDocument>(
+                RawBsonDocumentHelper.FromJson(
+                    "{conversationId: 1, payload: BinData(0,\"dj1VTVdlSTI1SkQxeU5ZWlJNcFo0Vkh2aFo5ZTA9\"), done: true, ok: 1}"));
+
+            var connection = new MockConnection(__serverId);
+            connection.EnqueueReplyMessage(saslStartReply);
+            connection.EnqueueReplyMessage(saslContinueReply);
+
+            Action act;
+            if (async)
+            {
+                act = () => subject.AuthenticateAsync(connection, __description, CancellationToken.None).GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                act = () => subject.Authenticate(connection, __description, CancellationToken.None);
+            }
+
+            act.ShouldNotThrow();
+            SpinWait.SpinUntil(() => connection.GetSentMessages().Count >= 2, TimeSpan.FromSeconds(5)).Should()
+                .BeTrue();
+
+            subject._cache().Should().NotBe(null);
+            subject._cache()._cacheKey().Should().NotBe(null);
+            subject._cache()._cacheValue().Should().NotBe(null);
```

Comment:
Didn't you rename `_cacheValue` to `_cachedEntry`?

Reply:
Yes, my bad. Fixed now.